### DISCUSSION TITLE
Why do one need to add +1 to register to get the location?

### DISCUSSION THREAD

**Discussion by VRciF** [yozik04/nibe#93] [Q&A] [answered]
Hi all,
I just started reading values from my new heatpump S2125-12 with an SMO S40 via modbus over TCP/IP. The csv export from the SMO S40 has slightly differing values compared to the json, but the csv matches the one in the github repo (https://github.com/yozik04/nibe/blob/master/nibe/data/smos40.csv).  

Since there is a separate [commit](https://github.com/yozik04/nibe/commit/17eef3ac54e5fa383fb198d4620d934805065bcb) for adjusting the offset for S320 - which is another model, right? - I'm just confused if some values I get are actually correct or just the values from the "next" register due to the +1 from the commit.

Why is this +1 needed? Is this just how modbus does it or is this something nibe specific? Does or can this change depending on future supported heat pumps or was this a needed fix for `convert_csv.py` in general? Maybe this is a question for the author @elupus ?

Many thanks for your time and many thanks for this great project!

**Comment by elupus** [comment, marked as answer]
It's modbus specific. That is how you write addresses. Silly yes. But that is the standard.

That said i think we should change how we name the registers for the s series for display purposes to match docs better.

**Comment by elupus** [comment]
If you want some details: https://control.com/forums/threads/modbus-register-numbering.49844/
